docs/config.md

```diff
@@ -94,6 +96,7 @@ Tekton Chains supports the following docstore services:
 | Key | Description | Supported Values | Default |
 | :--- | :--- | :--- | :--- |
 | `builder.id` | The builder ID to set for in-toto attestations | | `https://tekton.dev/chains/v2`|
+| `builder.type` | The builder type to set for in-toto attestations | | `https://tekton.dev/attestations/chains@v2`|
```

Comment:
Do we actually want this to be configurable at the configmap level? Based on https://slsa.dev/provenance/v0.2, the description of this field is:

```
URI indicating what type of build was performed. It determines the meaning of invocation, buildConfig and materials.
```

I'd expect this to identity the data included in the provenance as a TaskRun or PipelineRun

This raises an interesting question though - the current default value may not actually be correct here (particularly as we think about supporting pipelineruns) and we may want to change it. 🤔  cc @lcarva

Reply:
> Do we actually want this to be configurable at the configmap level?

I think making this field configurable will enable the builders who are built on the top of tekton to set this `builder.type` to whatever they want just like the `builder.id` field.

> I'd expect this to identity the data included in the provenance as a TaskRun or PipelineRun
> This raises an interesting question though - the current default value may not actually be correct here (particularly as we think about supporting pipelineruns) and we may want to change it. 🤔 cc @lcarva

Sounds interesting. Based on https://slsa.dev/provenance/v0.2, this field should be typeUri. Any idea on what the identifying uri should look like? Should it still be something like `https://tekton.dev/...`?

Reply:
> I think making this field configurable will enable the builders who are built on the top of tekton to set this `builder.type` to whatever they want just like the `builder.id` field.

It's a bit confusing though since the type we're setting in the intoto attestation is about the build (`buildType`), not the builder. I'd expect this to match the type we're attesting to, so I'm not sure if we want this to be configurable by service providers.
 
> > I'd expect this to identity the data included in the provenance as a TaskRun or PipelineRun
> > This raises an interesting question though - the current default value may not actually be correct here (particularly as we think about supporting pipelineruns) and we may want to change it. 🤔 cc @lcarva
> 
> Sounds interesting. Based on https://slsa.dev/provenance/v0.2, this field should be typeUri. Any idea on what the identifying uri should look like? Should it still be something like `https://tekton.dev/...`?

¯\_(ツ)_/¯ I think we have some flexibility here - maybe `tekton://<GroupVersionKind>`?

Reply:
Updated this to be `tekton.dev/v1beta1/kind=TaskRun`.

I didn't use [`GetGroupVersionKind().String()`](https://github.com/kubernetes/apimachinery/blob/v0.24.3/pkg/runtime/schema/group_version.go#L161-L163) directly since it will give us a string like `tekton.dev/v1beta1, Kind=TaskRun`.

Please have a look and let me know what you think.
Thanks

Reply:
Is `tekton.dev/v1beta1/kind=TaskRun` a valid URI? Not sure how strict this needs to be.

I also agree that this should not be configurable because it would imply the format of the intoto attestation is also configurable, which is certainly not the case today.

Reply:
> Is tekton.dev/v1beta1/kind=TaskRun a valid URI? Not sure how strict this needs to be.

Changed to `<group>/<version>/<kind>` i.e. tekton.dev/v1beta1/TaskRun. wdyt?


> I also agree that this should not be configurable because it would imply the format of the intoto attestation is also configurable, which is certainly not the case today.

Agree but interestingly Chains today allows `predicate.builder.id` to be [configurable](https://github.com/tektoncd/chains/blob/main/docs/config.md#in-toto-configuration).